crates/api_crud/src/post/update.rs

```diff
@@ -73,23 +73,33 @@ pub async fn update_post(
   }
 
   // Fetch post links and thumbnail if url was updated
-  let (embed_title, embed_description, embed_video_url, thumbnail_url) = match &url {
+  let (embed_title, embed_description, embed_video_url, metadata_thumbnail) = match &url {
     Some(url) => {
       let metadata = fetch_link_metadata(url, true, &context).await?;
       (
         Some(metadata.opengraph_data.title),
         Some(metadata.opengraph_data.description),
         Some(metadata.opengraph_data.embed_video_url),
-        Some(custom_thumbnail.map(Into::into).or(metadata.thumbnail)),
+        Some(metadata.thumbnail),
       )
     }
     _ => Default::default(),
   };
+
   let url = match url {
     Some(url) => Some(proxy_image_link_opt_apub(Some(url), &context).await?),
     _ => Default::default(),
   };
 
+  let custom_thumbnail = match custom_thumbnail {
+    Some(custom_thumbnail) => {
+      Some(proxy_image_link_opt_apub(Some(custom_thumbnail), &context).await?)
+    }
+    _ => Default::default(),
+  };
```

Comment:
Looks like this could be simply map.

Reply:
Can't unfortunately, like the one above it, async/await is unstable in maps.

Reply:
Right thats annoying